apps/src/applab/applab.js

```diff
@@ -776,6 +778,30 @@ Applab.init = function(config) {
   return loader;
 };
 
+function loadLibraryBlocks(config) {
+  if (!level.libraries) {
+    return;
+  }
+
+  level.libraryCode = '';
+  level.libraries.forEach(library => {
+    config.dropletConfig.additionalPredefValues.push(library.name);
+    level.libraryCode += library.source;
+    // TODO: add category management for libraries (blocked on spec)
+    // config.dropletConfig.categories['libraryName'] = {
+    //   id: 'libraryName',
+    //   color: 'colorName',
+    //   rgb: 'colorHexCode',
+    //   blocks: []
+    // };
+
+    library.dropletConfig.forEach(dropletConfig => {
+      config.dropletConfig.blocks.push(dropletConfig);
+      level.codeFunctions[dropletConfig.func] = null;
```

Comment:
why null?

Reply:
It's just the convention for adding a block to the toolbox:
https://github.com/code-dot-org/code-dot-org/blob/53f636676795c2bb9401326b6a108145747a1db2/lib/cdo/shared_constants.rb#L129